ml-agents-envs/mlagents_envs/registry/binary_utils.py

```diff
@@ -0,0 +1,154 @@
+import urllib.request
+import tempfile
+import os
+import uuid
+import shutil
+import glob
+
+from zipfile import ZipFile
+from sys import platform
+from typing import Tuple, Optional
+
+
+def get_local_binary_path(name: str, url: str) -> str:
+    """
+    Returns the path to the executable previously downloaded with the name argument. If
+    None is found, the executable at the url argument will be downloaded and stored
+    under name for future uses.
+    :param name: The name that will be given to the folder containing the extracted data
+    :param url: The URL of the zip file
+    """
+    path = get_local_binary_path_if_exists(name)
+    if path is None:
+        print(f"Local environment {name} not found, downloading environment from {url}")
+        download_and_extract_zip(url, name)
+    path = get_local_binary_path_if_exists(name)
+    if path is None:
+        raise FileNotFoundError("Binary not found")
+    return path
+
+
+def get_local_binary_path_if_exists(name: str) -> Optional[str]:
+    """
+    Recursively searches for a Unity executable in the extracted files folders. This is
+    platform dependent : It will only return a Unity executable compatible with the
+    computer's OS. If no executable is found, None will be returned.
+    """
+    _, bin_dir = get_tmp_dir()
```

Comment:
I think we should have the final location of these be in a user-overridable directory, not a temp dir (since ideally they won't be temp).

Reply:
I think tmp dir is fine for now. This is the way ai2-thor does it for example.